DICOM/src/Microsoft.Health.DeID.SharedLib/RedactFunction.cs

```diff
@@ -0,0 +1,147 @@
+        private static readonly int InitialDigitsCount = 3;
+        private readonly RedactSetting _redactSetting;
+
+        public RedactFunction(RedactSetting redactSetting = null)
+        {
+            _redactSetting = redactSetting ?? new RedactSetting();
+        }
+
+        public string RedactDate(string dateString, string inputDateFormat = null, string outputDateFormat = null, IFormatProvider provider = null)
+        {
+            if (string.IsNullOrEmpty(dateString))
+            {
+                return null;
+            }
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateString, inputDateFormat, provider, DeIDGlobalSettings.DateFormat);
+                return DateTimeUtility.IndicateAgeOverThreshold(date)
+                    ? null
+                    : new DateTimeOffset(date.Year, 1, 1, 0, 0, 0, new TimeSpan(0, 0, 0)).ToString(outputDateFormat ?? DeIDGlobalSettings.DateFormat);
+            }
+
+            return null;
+        }
+
+        public string RedactDateTime(string dateTimeString, string inputDateTimeFormat = null, string outputDateFormat = null, IFormatProvider provider = null)
+        {
+            if (string.IsNullOrEmpty(dateTimeString))
+            {
+                return null;
+            }
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                DateTimeOffset date = DateTimeUtility.ParseDateTimeString(dateTimeString, inputDateTimeFormat, provider);
+                return DateTimeUtility.IndicateAgeOverThreshold(date)
+                    ? null
+                    : new DateTimeOffset(date.Year, 1, 1, 0, 0, 0, new TimeSpan(0, 0, 0)).ToString(outputDateFormat ?? DeIDGlobalSettings.DateTimeFormat);
+            }
+
+            return null;
+        }
+
+        public DateTimeOffset? RedactDateTime(DateTimeOffset dateTime)
+        {
+            EnsureArg.IsNotNull<DateTimeOffset>(dateTime, nameof(dateTime));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateTime))
+                {
+                    return null;
+                }
+
+                return new DateTimeOffset(dateTime.Year, 1, 1, 0, 0, 0, default);
+            }
+
+            return null;
+        }
+
+        public DateTimeObject RedactDateTime(DateTimeObject dateObject)
+        {
+            EnsureArg.IsNotNull(dateObject, nameof(dateObject));
+
+            if (_redactSetting.EnablePartialDatesForRedact)
+            {
+                if (DateTimeUtility.IndicateAgeOverThreshold(dateObject.DateValue))
+                {
+                    return null;
+                }
+
+                dateObject.DateValue = new DateTimeOffset(dateObject.DateValue.Year, 1, 1, 0, 0, 0, dateObject.HasTimeZone == true ? dateObject.DateValue.Offset : new TimeSpan(0, 0, 0));
```

Comment:
🟡 **[Suggestion]**

`dateObject.HasTimeZone ? dataObject....` ? 😄

Reply:
Should be (bool)dateObject.HasTimeZone  or dateObject.HasTimeZone == true
Since dateObject.HasTimeZone is bool?